viite-backend/viite-main/src/main/scala/fi/liikennevirasto/viite/ProjectService.scala

```diff
@@ -1580,45 +1580,71 @@ def setCalibrationPoints(startCp: Long, endCp: Long, projectLinks: Seq[ProjectLi
         logger.warn(s"Ely-codes for project: $projectId were not updated.")
     }
 
+
+    /**
+     * Updates project links with data from the dev tool.
+     *
+     * This function takes the manually given devToolData, which contains the new address, roadway, and side code information,
+     * and updates the corresponding project links with these values. It also handles the creation of new
+     * roadway numbers if necessary, adjusts status and side codes, and sets calibration points for roadways.
+     *
+     * The function performs the following steps:
+     * 1. If both start and end address M values are defined, it updates the project links with these address values.
+     * 2. If none of the project links are marked as 'New', it updates the original address values.
+     * 3. If all project links are new, it generates a new roadway number for those links.
```

Comment:
How about, if some are new, some not? What then?

Reply:
Kyse on projektilinkkien tallennuksesta. Käyttäjä voi tallentaa vain yhtä toimenpidettä kerrallaan, eli kaikki on Uusia tai sitten ne ovat Siirtoja, tai Lakkautuksia etc.